### PR TITLE
pipx: minor refactor

### PR DESCRIPTION
**Pull request by russoz**
##### SUMMARY
Simplified the logic to determine `changed`

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
pipx

### CODE REVIEW

File: changelogs/fragments/3634-pipx-improve-changed.yaml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
```
Comment (felixfontein) [inline changelogs/fragments/3634-pipx-improve-changed.yaml:1]: In the PR you are saying it's a bugfix. Can you please elaborate whether this is a bugfix (if it is, please change this to `bugfixes:`) or not?

Comment (russoz) [reply changelogs/fragments/3634-pipx-improve-changed.yaml:1]: It was late last night when I submitted, guess I didn't think it through. Just changed to Feature PR instead.

Comment (russoz) [reply changelogs/fragments/3634-pipx-improve-changed.yaml:1]: It is not a bug. It is not a feature either, it is a minor improvement on the code, with no visible change to modules that use it.

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@russoz thanks for contributing this!

**Pull request merged**